$OpenBSD: patch-gio_gsocketconnection_c,v 1.2 2012/11/10 07:37:07 ajacoutot Exp $

Fix a reference counting bug (par rewrote upstream, remove when updating to glib >= 2.35).

--- gio/gsocketconnection.c.orig	Sat Nov 10 04:18:46 2012
+++ gio/gsocketconnection.c	Sat Nov 10 08:23:55 2012
@@ -201,6 +201,7 @@ g_socket_connection_connect_async (GSocketConnection  
     {
       g_simple_async_result_set_op_res_gboolean (simple, TRUE);
       g_simple_async_result_complete_in_idle (simple);
+      g_object_unref (simple);
     }
   else if (g_error_matches (tmp_error, G_IO_ERROR, G_IO_ERROR_PENDING))
     {
@@ -221,6 +222,7 @@ g_socket_connection_connect_async (GSocketConnection  
     {
       g_simple_async_result_take_error (simple, tmp_error);
       g_simple_async_result_complete_in_idle (simple);
+      g_object_unref (simple);
     }
 }
 
